src/main/java/com/crafter/springboottechiteasy/controllers/TelevisionController.java

```diff
@@ -2,56 +2,54 @@
 import com.crafter.springboottechiteasy.exceptions.RequirementsNotMetException;
+import com.crafter.springboottechiteasy.models.Television;
+import com.crafter.springboottechiteasy.repositories.TelevisionRepository;
+import com.crafter.springboottechiteasy.services.TelevisionService;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
+import java.util.List;
+
 @RestController
 @RequestMapping("/televisions")
 public class TelevisionController {
 
+    private TelevisionService televisionService;
+
+    public TelevisionController(TelevisionService televisionService) {
+        this.televisionService = televisionService;
+    }
+
     @GetMapping
-    public ResponseEntity<Object> getAllTVs() {
-        return ResponseEntity.ok("television");
+    public ResponseEntity<List<Television>> getAllTVs() {
+        return ResponseEntity.ok(televisionService.getAllTelevisions());
     }
 
     @GetMapping("/{id}")
-    public ResponseEntity<Object> getOneTV(@PathVariable int id) {
-        if (id == 0) { // determine if id exists
-            throw new IndexOutOfBoundsException();
-        } else {
-            return ResponseEntity.ok(id);
-        }
+    public ResponseEntity<Television> getOneTV(@PathVariable Long id) {
+        return ResponseEntity.ok(televisionService.getOneTelevision(id));
     }
 
     @PostMapping("/add")
-    public ResponseEntity<Object> createNewTV() {
-        String television = "Panasonic";
-        if (television.length() > 20) {
-            throw new RequirementsNotMetException("The name is too long. There is a max of 20 characters");
-        } else {
-            return ResponseEntity.created(null).body("television");
-        }
+    public ResponseEntity<Television> createNewTV(@RequestBody Television t) {
+        return ResponseEntity.created(null).body(televisionService.createTelevision(t));
     }
 
-    @PutMapping("/update/{id}")
-    public ResponseEntity<Object> updateTV(@PathVariable int id, @RequestBody String televisionName) {
-        if (id == 0) {
-            throw new RecordNotFoundException();
-        } else {
 
-            return ResponseEntity.noContent().build();
-        }
+    @PutMapping("/update/{id}")
+    public ResponseEntity<Television> updateTV(@PathVariable Long id, @RequestBody Television tv) {
+        televisionService.updateTelevision(id, tv);
+        return ResponseEntity.noContent().build();
     }
 
+
     @DeleteMapping("/delete/{id}")
     @ResponseStatus(code = HttpStatus.NO_CONTENT)
-    public ResponseEntity<Object> deleteTV(@PathVariable int id) {
-        if (id == 0) {
-            throw new RecordNotFoundException();
-        } else {
-            return ResponseEntity.noContent().build();
-        }
+    public ResponseEntity<Television> deleteTV(@PathVariable Long id) {
+        televisionService.deleteTelevision(id);
+        return ResponseEntity.noContent().build();
```

Comment:
Deze methode heeft 2 keer een status. Namelijk, via de ResponseEntity.noContent() en via de `@`ResponseStatus annotatie hier boven. Waarom doe je dit dubbelop? Wat gebeurt er als beide varianten (perongeluk) niet dezelfde status zouden terug geven? Heeft een van de twee dan voorrang? Als je nieuwschierig bent wat het antwoord is, dan kun je het eens uitproberen.